daft/context.py

```diff
@@ -89,7 +94,7 @@ def _get_runner_config_from_env() -> _RunnerConfig:
         raise ValueError(f"Unsupported DAFT_RUNNER variable: {runner_from_envvar}")
 
     # Retrieve the runner from current initialized Ray environment, only if not running in a Ray worker
-    elif ray_is_initialized and not in_ray_worker:
+    elif not in_ray_worker and (ray_is_initialized or ray_is_in_job):
```

Comment:
Wait please remind me again. Is it possible for this `elif` condition to be true? Namely, is it possible for `not in_ray_worker and ray_is_in_job` to be true? I thought that Ray jobs would never run on the head node (since the head node is purely for scheduling).

If `not in_ray_worker and ray_is_in_job` can never be true, then `not in_ray_worker and (ray_is_initialized or ray_is_in_job)` == `not in_ray_worker and ray_is_initialized`, which is the original condition prior to your changes...

Maybe I'm confusing myself here.

Reply:
Nevermind, resolved this with @kevinzwang offline! I was incorrect. Confusions have been cleared.